build: Update project dependencies and configurations

angel-penchev opened the pull request:
**Before opening a PR**
Please ensure your pull request adheres to the following guidelines:

- [x] Search previous pull requests before making a new one, as yours may be a duplicate.
- [x] Test and validate the new changes you add.
- [ ] Add test cases for any new code you introduce.
- [ ] Document your newly added changes.
- [x] Add the appropriate labels to your pull request.

**Referenced issue**
This pull request resolves #4.

**Change Description**
- Updated packages to their latest compatible versions.
- Migrated Nx configurations to be compatible with Nx v20.
- Reconfigured ESLint as `.eslintrc` files are now deprecated. Modified the rule set and resolved all pending linting changes.
- Applied prettier formatting.
- Updated Next.js cookies API usage as it was modified as part of Next.js v15.
- Updated internationalization configurations in both `fmicodes-admin` and `fmicodes-site`, as the `next-intl` library has modified its' configuration in the latest version.
- Changed husky usage to comply with the deprecation of the `husky install` command.
- Applied a pending DB migration.
- Modified method names in API controllers to remove `V1` from their names, as the client generation was producing function names with double V1s as such - `V1V1`.
- Resolved TailwindCSS upgrade errors.
- Added explicit hint for nvm to use Node.js v22 LTS.

**Root Cause (if fixing a bug)**
N/A

**Breaking change?**
Is the change you are introducing going to affect already other existing resources? (ex. addition of a new required field for an API endpoint may require additional WEB client changes):

- [ ] No, this code change does not affect other project resources.
- [x] Yes, the code changes in this pull request modify already existing project resources.

**Additional context**
I am not doing that again.

angel-penchev committed:
build: update dependency versions

Updated the dependency packages versions to their latest available
versions.

angel-penchev committed:
chore: apply pending db migration

angel-penchev committed:
chore: update husky usage

Updated prepare script to use updated husky command.

angel-penchev committed:
build: specify node version for nvm

angel-penchev committed:
chore: adjust linter packages names

Corrected linter package names after package.json update.

angel-penchev committed:
build: bump project deps

angel-penchev committed:
build: remove deprecated isolated config flag

angel-penchev committed:
fix: correct ms usage type

angel-penchev committed:
build: adjust package versions for nx

Adjusted several package versions for compatability with Nx.

angel-penchev committed:
build: update next cookies usage after v15 upgrade

Updated all places cookies() method is called in Next.js projects to
account for the new way the API works in Next.js v15.

angel-penchev committed:
refactor: update next-intl usage in fmicodes-site

As the next-intl package has its' dependencies refactored, I have
updated how it is being used.

angel-penchev committed:
chore: eslint mjs conversion

angel-penchev committed:
refactor: remove v1 for api method names

Removed "V1" from API controler names.

angel-penchev committed:
fix: correct serve command

angel-penchev committed:
style: apply prettier formatting

angel-penchev committed:
style: reconfigure linting and apply it

Resolved all eslint errors and applied linting to the entire repository.
(fking finally)

angel-penchev committed:
fix: resolve fmicodes-site build errors

Resolved all fmicodes-site compilation/linting errors.

angel-penchev committed:
refactor: update next-intl usage in fmicodes-admin

Updated internationalization configuration in fmicodes-admin as the
library modified how it is being used.

angel-penchev committed:
fix: tailwind configuration in fmicodes-admin

Resolved an issue where TailwindCSS styles were not being applied in the
fmicodes-admin project.

angel-penchev committed:
chore: remove a redundant .gitkeep file

Removed a redundant .gitkeep file from the libs/ directory.

MishoMish commented on apps/fmicodes-admin-e2e/src/support/commands.ts:
```diff
@@ -10,7 +10,6 @@
 
 // eslint-disable-next-line @typescript-eslint/no-namespace
 declare namespace Cypress {
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
```
And we are starting to use it somewhere, or the ignore was simply forgotten?

MishoMish replied:
Got it ... its ok

MishoMish commented on apps/fmicodes-admin/i18n/routing.ts:
```diff
@@ -0,0 +1,9 @@
+  locales: ['en', 'bg'],
+
+  // Used when no locale matches
+  defaultLocale: 'en',
```
The default for us is **en** or **bg**? Will look this up later but probably **bg** will be the one. The **en** can be used if needed.

angel-penchev replied:
My rationale last year was that if someone's machine/browser/etc. is set to Bulgarian, they prefer things to be displayed in Bulgarian - otherwise, default to English for all other locales. I am open to changing it; however, just confirm what is the required behaviour. :))

MishoMish commented on apps/fmicodes-admin/next-env.d.ts:
```diff
@@ -2,4 +2,4 @@
 
 // NOTE: This file should not be edited
-// see https://nextjs.org/docs/basic-features/typescript for more information.
+// see https://nextjs.org/docs/app/api-reference/config/typescript for more information.
```
![image](https://github.com/user-attachments/assets/27247b09-2e9e-4e06-acf7-ae45739b44a6)

angel-penchev replied:
Next decided to break its own rule during the upgrade. ¯\_(ツ)_/¯

MishoMish reviewed (approved):
Went briefly throughout the last 4/5ths of the files. Most of them i see are linter modifications because of the new versions. Don't mind the other comments - count it as approved ;d

angel-penchev committed:
build: switch from npm to pnpm in container packaging

angel-penchev committed:
chore: add .gitattributes file

Add a .gitattributes file, specifying that the project should use lf
line endings.

angel-penchev committed:
ci: temporary disable automated deployment

angel-penchev committed:
ci: add generated client files to prettier ignore list

angel-penchev committed:
test: adjust team test expectation

Adjusted team test create() expectation to include project fields.

angel-penchev committed:
test: proper redis mock in users service test

Added Redis mocking in the UsersService.

angel-penchev committed:
fix: next-intl runtime error in fmicodes-admin

Resolved a runtime error in fmicodes-admin caused due to the next-intl
package upgrade.
